system/lib/libc/emscripten_memset.c

```diff
@@ -9,9 +9,16 @@ __attribute__((no_sanitize("address"))) void *__memset(void *str, int c, size_t
 __attribute__((__weak__)) void *__musl_memset(void *str, int c, size_t n);
 __attribute__((__weak__)) void *__memset(void *str, int c, size_t n);
 
-#ifdef EMSCRIPTEN_OPTIMIZE_FOR_OZ
+#if defined(EMSCRIPTEN_OPTIMIZE_FOR_OZ) || __has_feature(address_sanitizer)
 
 void *__memset(void *str, int c, size_t n) {
+#if !defined(EMSCRIPTEN_STANDALONE_WASM)
+  if (n >= 512) {
+    _emscripten_memset_js(str, c, n);
+    return str;
+  }
+#endif
```

Comment:
It looks like [`emscripten_memcpy.c`](https://github.com/emscripten-core/emscripten/blob/efdfe40d37eadd258e7c1751bd42209e0915c40c/system/lib/libc/emscripten_memcpy.c) only uses the `_js` version once in that file, in the last case. That is, it doesn't use it in the first case, here, for `-Oz`/ASan. Is there a reason to do things differently for memset?

Reply:
We want good performance in `-Oz`, right? I can understand not wanting it to apply for asan, though, since the JS fill will bypass asan checks.

Reply:
Yeah, for ASan I think we want to be able to instrument it as you said, and performance is less critical there.

For `-Oz` we've focused on reducing size at all costs, which includes avoiding loop unrolling here. I think avoiding adding a fast path through JS is consistent with that?